4-custom-evaluations.ipynb

```diff
@@ -0,0 +1,744 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Monitoring your RAG Applications using LLM Observability & RAGAS\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Datadog's LLM Observability tool allows you to perform evaluations on your LLM Application and tie these evaluations to specific traces. \n",
+    "\n",
+    "In the following example, we'll build & trace a RAG-powered application using [Llamaindex](https://docs.llamaindex.ai/en/latest/) that answers all your LLM Observability onboarding questions. We'll plug in retrieval and augmented generation strategies to our RAG app and evaluate these different strategies using [RAGAS](https://docs.ragas.io/en/stable/concepts/index.html).\n",
+    "\n",
+    "#### Learning Goals\n",
+    "- Understand how to instrument a RAG-powered LLM App\n",
+    "- Understand how to submit custom evaluations tied to specific traces for your LLM App"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setup & Pre-requisites"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "First, make sure you've followed the setup directions in the README. Then, install Llamaindex, which we'll be using to build our RAG workflow."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m23.0.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m24.0\u001b[0m\n",
+      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
```

Comment:
```suggestion
      "Note: You may need to restart the kernel to use updated packages.\n"
```

Reply:
I believe this is an autogenerated message from pip downloads, will just clear it from the cell output